portfolio/src/main/webapp/script.js

```diff
@@ -16,13 +16,46 @@
  * Adds a random greeting to the page.
  */
 function addRandomGreeting() {
-  const greetings =
-      ['The Setesh guard’s nose drips.', 'La nariz del guardia de Setesh gotea.', '塞特什警卫队的鼻子在滴水','Le nez du garde Seth dégoulinait.'];
+  fetch('/data').then(response => response.json()).then((data) => {
+    const greeting = data[Math.floor(Math.random()*data.length)];
+    const greetingContainer = document.getElementById('greeting-container');
+    greetingContainer.innerText = greeting;
+  });
+}
+
+function getData(){
+  const name = document.getElementById('text-input').value; 
+  fetch('/data2?name='+name).then(response => response.text()).then((data) => {
```

Comment:
Let's learn about string interpolation:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

you can make this cleaner by writing

`
fetch('/data2?name=${name}').then(response => ...
`

Reply:
Fixed, I also inlined getting the value